velox/vector/tests/VectorTest.cpp

```diff
@@ -3033,5 +3033,21 @@ TEST_F(VectorTest, mutableValues) {
   }
 }
 
+TEST_F(VectorTest, mutableIndices) {
+  // Ensure that DictionaryVector::mutableIndices always returns a new indices
+  // buffer if its not unique.
+  auto flatVector = makeFlatVector<int64_t>({1, 2, 3, 4, 5, 6, 7, 8});
+  auto indices = makeIndices(8, [](auto row) { return row % 2; });
+  auto vector = BaseVector::wrapInDictionary(nullptr, indices, 8, flatVector);
+  ASSERT_EQ(vector->encoding(), VectorEncoding::Simple::DICTIONARY);
+  auto dictionary = vector->as<DictionaryVector<int64_t>>();
+
+  auto mutableIndices = dictionary->mutableIndices(1);
```

Comment:
Let's also verify that mutableIndices is different from indices.

Let's also add test case when mutableIndices returns the same buffer.